core/mutator/entry/entry_test.go

```diff
@@ -71,18 +72,21 @@ func createEntry(commitment []byte, pkeys []string) ([]byte, error) {
 		}
 	}
 
-	entry := &tpb.Entry{
+	return &tpb.Entry{
 		Commitment:     commitment,
 		AuthorizedKeys: authKeys,
-	}
-	entryData, err := proto.Marshal(entry)
-	if err != nil {
-		return nil, fmt.Errorf("Marshal(%v)=%v", entry, err)
-	}
-	return entryData, nil
+	}, nil
 }
 
-func prepareMutation(key []byte, entryData []byte, previous []byte, signers []signatures.Signer) ([]byte, error) {
+func prepareMutation(key []byte, newEntry *tpb.Entry, previous []byte, signers []signatures.Signer) (*tpb.SignedKV, error) {
+	var entryData []byte
+	if newEntry != nil {
```

Comment:
It should not be valid to set `newEntry` with `nil`. Please return an error in this case.